java/code/src/com/redhat/rhn/common/localization/XmlResourceBundle.java

```diff
@@ -84,26 +87,27 @@ public XmlResourceBundle(String filelocation) throws IOException {
             // access to it over the web when starting up the service.
             String xsdLocation =
                 this.getClass().getResource("/xliff-core-1.1.xsd").toString();
-            XMLReader parser = XMLReaderFactory.
-                createXMLReader("org.apache.xerces.parsers.SAXParser");
-            parser.setFeature(validationFeature, false);
-            parser.setFeature(schemaFeature, false);
-            parser.setProperty("http://apache.org/xml/properties/schema/" +
+            SAXParserFactory factory = SAXParserFactory.newInstance();
+            factory.setFeature(validationFeature, false);
+            factory.setFeature(schemaFeature, false);
+            factory.setFeature("http://xml.org/sax/features/external-general-entities", false);
+            factory.setFeature("http://xml.org/sax/features/external-parameter-entities", false);
+            SAXParser parser = factory.newSAXParser();
+            XMLReader reader = parser.getXMLReader();
+            reader.setProperty("http://apache.org/xml/properties/schema/" +
                     "external-noNamespaceSchemaLocation", xsdLocation);
             XmlResourceBundleParser handler = new XmlResourceBundleParser();
-            parser.setContentHandler(handler);
-            parser.parse(new InputSource(this.getClass().
+            reader.setContentHandler(handler);
+            reader.parse(new InputSource(this.getClass().
                                          getResourceAsStream(filelocation)));
             strings = handler.getMessages();
```

Comment:
I had to read a bit why the `SAXParserFactory` should be used over `XMLReaderFactory`. It seems perfectly logical as far as I can tell

Reply:
the old one has been deprecated in Java 9